libs/libc/modlib/modlib_load.c

```diff
@@ -271,16 +293,29 @@ int modlib_load(FAR struct mod_loadinfo_s *loadinfo)
         }
     }
 #else
-  loadinfo->textalloc = (uintptr_t)lib_malloc(loadinfo->textsize +
```

Comment:
Why go through the different patch with or without CONFIG_ARCH_USE_MODULE_TEXT?

Reply:
just because the existing code is different.